program-memory/src/lib.rs

```diff
@@ -229,7 +223,7 @@ mod tests {
         for dst in 13..20 {
             assert!(is_nonoverlapping(10, 3, dst, 3));
         }
-        assert!(is_nonoverlapping::<u8>(255, 3, 254, 1));
-        assert!(!is_nonoverlapping::<u8>(255, 2, 254, 3));
+        assert!(is_nonoverlapping(255, 3, 254, 1));
+        assert!(!is_nonoverlapping(255, 2, 254, 3));
```

Comment:
Since this test is meant to look at the limits, can it be updated to work with `usize` limits?
```suggestion
        assert!(is_nonoverlapping(usize::MAX, 3, usize::MAX - 1, 1));
        assert!(!is_nonoverlapping(usize::MAX, 2, usize::MAX - 1, 3));
```